integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/automaticindexing/ApplicationAutomaticIndexingFilterIT.java

```diff
@@ -0,0 +1,273 @@
+				ctx -> ctx.exclude( EntityA.class )
+						.include( Entity1A.class )
+		);
+		setupHolder.runInTransaction( session -> {
+			session.persist( new EntityA( 10, "test" ) );
+			session.persist( new Entity1A( 20, "test" ) );
+			session.persist( new Entity1B( 30, "test" ) );
+			session.persist( new Entity2A( 40, "test" ) );
+
+			backendMock.expectWorks( Entity1A.INDEX )
+					.add( "20", b -> b.field( "indexedField", "test" ) );
+			backendMock.expectWorks( Entity2A.INDEX )
+					.add( "40", b -> b.field( "indexedField", "test" ) );
+		} );
+		backendMock.verifyExpectationsMet();
+
+		// only include - should include all since no excludes.
+		Search.automaticIndexingFilter(
+				setupHolder.entityManagerFactory(),
+				ctx -> ctx.include( Entity1A.class )
+		);
+		setupHolder.runInTransaction( session -> {
+			session.persist( new EntityA( 100, "test" ) );
+			session.persist( new Entity1A( 200, "test" ) );
+			session.persist( new Entity1B( 300, "test" ) );
+			session.persist( new Entity2A( 400, "test" ) );
+
+			backendMock.expectWorks( EntityA.INDEX )
+					.add( "100", b -> b.field( "indexedField", "test" ) );
+			backendMock.expectWorks( Entity1A.INDEX )
+					.add( "200", b -> b.field( "indexedField", "test" ) );
+			backendMock.expectWorks( Entity1B.INDEX )
+					.add( "300", b -> b.field( "indexedField", "test" ) );
+			backendMock.expectWorks( Entity2A.INDEX )
+					.add( "400", b -> b.field( "indexedField", "test" ) );
+		} );
+		backendMock.verifyExpectationsMet();
+	}
+
+	@Test
+	public void sameClassFails() {
+		assertThatThrownBy( () ->
+				Search.automaticIndexingFilter(
+						setupHolder.entityManagerFactory(),
+						ctx -> ctx.exclude( EntityA.class )
+								.include( EntityA.class )
+				)
+		).isInstanceOf( SearchException.class )
+				.hasMessageContainingAll(
+						EntityA.class.getName(),
+						"cannot be included and excluded at the same time within one filter",
+						"Already included types: '[]'",
+						"Already excluded types:"
+				);
+
+		assertThatThrownBy( () ->
+				Search.automaticIndexingFilter(
+						setupHolder.entityManagerFactory(),
+						ctx -> ctx.include( EntityA.class )
+								.exclude( EntityA.class )
+				)
+		).isInstanceOf( SearchException.class )
+				.hasMessageContainingAll(
+						EntityA.class.getName(),
+						"cannot be included and excluded at the same time within one filter",
+						"Already included types:",
+						"Already excluded types: '[]'"
+				);
+	}
+
+	@Test
+	public void sameNameFails() {
+		assertThatThrownBy( () ->
+				Search.automaticIndexingFilter(
+						setupHolder.entityManagerFactory(),
+						ctx -> ctx.include( EntityA.INDEX )
+								.exclude( EntityA.INDEX )
+				)
+		).isInstanceOf( SearchException.class )
+				.hasMessageContainingAll(
+						EntityA.class.getName(),
+						"cannot be included and excluded at the same time within one filter",
+						"Already included types:",
+						"Already excluded types: '[]'"
+				);
+	}
```

Comment:
Good point! Most of these cases are tested in the session filter test. These two here are just to make sure that app filter throws exceptions too. Since app/session filters are the same impl I've only kept these two tests here. I'll add a test for a string-that-is-not-a-name, for class -- there're a few already.